### DISCUSSION TITLE
Integrating PCIe in the Large Data Mode of Fast-DDS

### DISCUSSION THREAD

**Discussion by AnoopGQuest** [eProsima/Fast-DDS#4919] [General]
Currently in the large data mode, Fast DDS discovers all nodes via UDP and later changes the data transfer to TCP, leveraging TCP's benefits of transferring data and avoiding manual configuration complexities. Can we introduce a transport protocol PCIe and use that PCIe in the large data configuration of Fast- DDS?

Which means UDP for discovery and PCIe for data transfer...

**Comment by JesusPoderoso** [comment]
Hi @AnoopGQuest, 
these thread discussions (#4623 and #4551) may interest you.
At this moment, there is no PCIe transport, and it is not part of our short-term roadmap.
Introducing a PCIe transport implies a custom solution (implement or update the Transport interface, Transport descriptor, SenderResource, Channel Resource, and XML Parser  classes). If you need more help in relation to that, do not hesitate contacting our commercial support: [support@eprosima.com](mailto:support@eprosima.com)
